### PR TITLE
[DOCS] Edit ILM GS tutorial

### PR DESCRIPTION
**Pull request by debadair**

### CODE REVIEW

File: docs/reference/ilm/getting-started-ilm.asciidoc [hunk 1]
```diff
@@ -3,77 +3,87 @@
...
-of 50 gigabytes, or has been created 30 days ago, and then delete the index
-after 90 days.
+This tutorial demonstrates how to use {ilm} ({ilm-init}) 
+to manage indices that contain time-series data. 
+
+When you use a mechanism like Filebeat to continuously index documents into {es},
+you typically use an index alias so you can periodically roll over to a new index.
+This enables you to implement a hot-warm-cold architecture to meet your performance
+requirements for your newest data, control costs over time, enforce retention policies,
+and still get the most out of your data.
```
Comment (andreidan) [inline docs/reference/ilm/getting-started-ilm.asciidoc]: can we emphasize that any data source is ok to be used? (I think the current version, albeit more composed, ties ILM to Filebeat, which is something that some users might not be familiar with) Maybe something along the lines of
"When using a a time series data source like Filebeat to continuously ..." ?

File: docs/reference/ilm/getting-started-ilm.asciidoc [hunk 2]
```diff
@@ -3,77 +3,87 @@
...
+
+To automate roll over and management of time-series indices with {ilm-init}, you:
+
+. <<ilm-gs-create-policy, Create a lifecycle policy>> with the {ilm-init} Put policy API.  
+. <<ilm-gs-apply-policy, Create an index template>> to apply the policy to each new time-series index.
+. <<ilm-gs-bootstrap, Bootstrap the initial time-series index>> by explicitly designating it as the write index.
+. <<ilm-gs-check-progress, Verify that indexes are moving through the lifecycle phases>> as expected with the 
+you use the {ilm-init} explain API.
```
Comment (andreidan) [inline docs/reference/ilm/getting-started-ilm.asciidoc]: "with the explain API"

File: docs/reference/ilm/getting-started-ilm.asciidoc [hunk 3]
```diff
@@ -3,77 +3,87 @@
...
 [[ilm-gs-create-policy]]
-=== Setting up a policy
+=== Create a lifecycle policy
 
-There are many new features introduced by {ilm-init}, but we will only focus on
-a few that are needed for our example. For starters, we will use the
-<<ilm-put-lifecycle,Put Policy>> API to define our first policy. Lifecycle
-policies are defined in JSON and include specific
-<<ilm-policy-definition,phases and actions>>.
+In your lifecycle policy, you specify the phases in your index lifecycle
```
Comment (andreidan) [inline docs/reference/ilm/getting-started-ilm.asciidoc]: Shall we rephrase "In your lifecycle policy" to "When you create a lifecycle policy" ? ("in your ..." sounds a bit confusing as the user is just looking to create a policy now)

File: docs/reference/ilm/getting-started-ilm.asciidoc [hunk 4]
```diff
@@ -3,77 +3,87 @@
...
+and the actions to perform in each phase. A lifecycle can have up to four  phases:
+`hot`, `warm`, `cold`, and `delete`. Policies are defined in JSON 
+and added through the {ilm-init} Put policy API. 
+
+For example, the following request creates a `data_stream` policy with two phases:
+ 
+* The `hot` phase defines a `rollover` action to specify that an index rolls over when it 
+reaches either a `max_size` of 50 gigabytes or a `max_age` of 30 days.
+* The `delete` phase uses `min_age` to move an index into the
+delete phase when it is 90 days old and immediately triggers the  `delete` action to remove the index.
```
Comment (andreidan) [inline docs/reference/ilm/getting-started-ilm.asciidoc]: nit: "the  `delete` action" - there're some extra spaces before `delete`

File: docs/reference/ilm/getting-started-ilm.asciidoc [hunk 5]
```diff
@@ -3,77 +3,87 @@
...
-<3> the hot phase defined in the "phases" section. Optional `min_age` field
-    not defined -- defaults to `0ms`
-<4> rollover action definition
-<5> delete phase begins after 90 days
-<6> delete action definition
 
+<1> The `min_age` field defaults to `0ms` in the hot phase, so new indices immediately enter the hot phase. 
+<2> The `rollover` action is triggered when any of the conditions are met.
+<3> An index is moved into the delete phase after 90 days.
+<4> No additional conditions are specified, so the `delete` action is triggered when the index enters the delete phase.
```
Comment (andreidan) [inline docs/reference/ilm/getting-started-ilm.asciidoc]: the delete action doesn't support any conditionals/parameters so I think we shouldn't mention conditionals here but rather "The delete action is triggered ..."

File: docs/reference/ilm/getting-started-ilm.asciidoc [hunk 6]
```diff
@@ -90,22 +100,19 @@ PUT _template/datastream_template
...
-that all newly created indices prefixed "datastream-" will be managed by
-our policy. The other setting used here is `index.lifecycle.rollover_alias`.
-This setting is required when using a policy containing the rollover
-action and specifies which alias to rollover on behalf of this index.
-The intention here is that the rollover alias is also defined on the index.
-
-To begin, we will want to bootstrap our first index to write to.
+<1> Apply the template to a new index if its name starts with "datastream-". 
+<2> The name of the lifecycle policy to apply to each new index.
+<3> The index alias must be specified if a policy uses the rollover action.
```
Comment (andreidan) [inline docs/reference/ilm/getting-started-ilm.asciidoc]: nit: "The rollover alias"

**Review by andreidan — CHANGES_REQUESTED** [review]
Thanks for working on this @debadair 

I think this will be a great improvement and it'll help make the "join the ilm train" experience smoother.

I've left a few suggestions below.

File: docs/reference/ilm/getting-started-ilm.asciidoc (hunk 1 above)
Comment (debadair) [reply docs/reference/ilm/getting-started-ilm.asciidoc]: Changed to, "When you continuously index timestamped documents into {es} using 
Filebeat, Logstash, or some other mechanism..."

File: docs/reference/ilm/getting-started-ilm.asciidoc (hunk 4 above)
Comment (debadair) [reply docs/reference/ilm/getting-started-ilm.asciidoc]: Fixed.

File: docs/reference/ilm/getting-started-ilm.asciidoc (hunk 5 above)
Comment (debadair) [reply docs/reference/ilm/getting-started-ilm.asciidoc]: Fixed.

**Comment by debadair** [comment]
@elasticmachine update branch

File: docs/reference/ilm/getting-started-ilm.asciidoc [hunk 7]
```diff
@@ -100,23 +119,17 @@ DELETE /_template/datastream_template
...
+[[ilm-gs-bootstrap]]
+=== Bootstrap the initial time-series index
 
-To begin, we will want to bootstrap our first index to write to.
+To get things started, you need to bootstrap an initial index and 
+designate it as the write index for the rollover alias specified in your index template. 
+The name of this index must match the template's index pattern and end with a number. 
+On rollover, this value is incremented to generate a name for the new index. 
 
+For example, the following request creates an index called `datastream_000001`
```
Comment (andreidan) [inline docs/reference/ilm/getting-started-ilm.asciidoc]: `datastream_000001` -> `datastream-000001`

**Review by andreidan — APPROVED** [review]
Thanks for iterating on this @debadair. This is great! There's one typo in the index name , but other than that LGTM

File: docs/reference/ilm/getting-started-ilm.asciidoc [hunk 8]
```diff
@@ -1,79 +1,93 @@
 [role="xpack"]
 [testenv="basic"]
+
 [[getting-started-index-lifecycle-management]]
-== Get started with {ilm}
+== Get started: Automate roll over with {ilm-init}
```
Comment (jrodewig) [inline docs/reference/ilm/getting-started-ilm.asciidoc]: I think it's `rollover` when used as a noun.
```suggestion
== Get started: Automate rollover with {ilm-init}
```

File: docs/reference/ilm/getting-started-ilm.asciidoc [hunk 9]
```diff
@@ -1,79 +1,93 @@
...
+
+++++
+<titleabbrev>Automate roll over</titleabbrev>
```
Comment (jrodewig) [inline docs/reference/ilm/getting-started-ilm.asciidoc]:
```suggestion
<titleabbrev>Automate rollover</titleabbrev>
```

File: docs/reference/ilm/getting-started-ilm.asciidoc [hunk 10]
```diff
@@ -1,79 +1,93 @@
...
+++++
+
+This tutorial demonstrates how to use {ilm} ({ilm-init}) 
+to manage indices that contain time-series data. 
+
+When you continuously index timestamped documents into {es} using 
+Filebeat, Logstash, or some other mechanism, 
+you typically use an index alias so you can periodically roll over to a new index.
```
Comment (jrodewig) [inline docs/reference/ilm/getting-started-ilm.asciidoc]: Link for index alias?

File: docs/reference/ilm/getting-started-ilm.asciidoc [hunk 11]
```diff
@@ -1,79 +1,93 @@
...
+This enables you to implement a hot-warm-cold architecture to meet your performance
+requirements for your newest data, control costs over time, enforce retention policies,
+and still get the most out of your data.
 
-Let's jump into {ilm} ({ilm-init}) by working through a hands-on scenario.
-This section will leverage many new concepts unique to {ilm-init} that
-you may not be familiar with. The following sections will explore
-these in more details.
+To automate roll over and management of time-series indices with {ilm-init}, you:
```
Comment (jrodewig) [inline docs/reference/ilm/getting-started-ilm.asciidoc]:
```suggestion
To automate rollover and management of time-series indices with {ilm-init}, you:
```

File: docs/reference/ilm/getting-started-ilm.asciidoc [hunk 12]
```diff
@@ -1,79 +1,93 @@
...
 
-The goal of this example is to set up a set of indices that will encapsulate
-the data from a time series data source. We can imagine there is a system
-like {filebeat-ref}[Filebeat] that continuously indexes documents into
-our writing index. We wish to roll over the index after it reaches a size
-of 50 gigabytes, or has been created 30 days ago, and then delete the index
-after 90 days.
+. <<ilm-gs-create-policy, Create a lifecycle policy>> with the {ilm-init} Put policy API.
```
Comment (jrodewig) [inline docs/reference/ilm/getting-started-ilm.asciidoc]: I don't think "Put" needs to capitalized.
> . <<ilm-gs-create-policy, Create a lifecycle policy>> with the {ilm-init} Put policy API.

File: docs/reference/ilm/getting-started-ilm.asciidoc [hunk 13]
```diff
@@ -1,79 +1,93 @@
...
-The goal of this example is to set up a set of indices that will encapsulate
-the data from a time series data source. We can imagine there is a system
-like {filebeat-ref}[Filebeat] that continuously indexes documents into
-our writing index. We wish to roll over the index after it reaches a size
-of 50 gigabytes, or has been created 30 days ago, and then delete the index
-after 90 days.
+. <<ilm-gs-create-policy, Create a lifecycle policy>> with the {ilm-init} Put policy API.  
+. <<ilm-gs-apply-policy, Create an index template>> to apply the policy to each new index.
+. <<ilm-gs-bootstrap, Bootstrap an index>> as the initial write index.
+. <<ilm-gs-check-progress, Verify that indices are moving through the lifecycle phases>>
```
Comment (jrodewig) [inline docs/reference/ilm/getting-started-ilm.asciidoc]:
```suggestion
. <<ilm-gs-check-progress, Verify indices are moving through the lifecycle phases>> 
```

File: docs/reference/ilm/getting-started-ilm.asciidoc [hunk 14]
```diff
@@ -1,79 +1,93 @@
...
-the data from a time series data source. We can imagine there is a system
-like {filebeat-ref}[Filebeat] that continuously indexes documents into
-our writing index. We wish to roll over the index after it reaches a size
-of 50 gigabytes, or has been created 30 days ago, and then delete the index
-after 90 days.
+. <<ilm-gs-create-policy, Create a lifecycle policy>> with the {ilm-init} Put policy API.  
+. <<ilm-gs-apply-policy, Create an index template>> to apply the policy to each new index.
+. <<ilm-gs-bootstrap, Bootstrap an index>> as the initial write index.
+. <<ilm-gs-check-progress, Verify that indices are moving through the lifecycle phases>> 
+as expected with the {ilm-init} Explain API.
```
Comment (jrodewig) [inline docs/reference/ilm/getting-started-ilm.asciidoc]:
```suggestion
as expected with the {ilm-init} explain API. 
```

File: docs/reference/ilm/getting-started-ilm.asciidoc [hunk 15]
```diff
@@ -1,79 +1,93 @@
...
-=== Setting up a policy
+=== Create a lifecycle policy
 
-There are many new features introduced by {ilm-init}, but we will only focus on
-a few that are needed for our example. For starters, we will use the
-<<ilm-put-lifecycle,Put Policy>> API to define our first policy. Lifecycle
-policies are defined in JSON and include specific
-<<ilm-policy-definition,phases and actions>>.
+A lifecycle policy specifies the phases in the index lifecycle
+and the actions to perform in each phase. A lifecycle can have up to four  phases:
```
Comment (jrodewig) [inline docs/reference/ilm/getting-started-ilm.asciidoc]:
```suggestion
and the actions to perform in each phase. A lifecycle can have up to four phases:
```

File: docs/reference/ilm/getting-started-ilm.asciidoc [hunk 16]
```diff
@@ -1,79 +1,93 @@
...
+`hot`, `warm`, `cold`, and `delete`. Policies are defined in JSON 
+and added through the {ilm-init} Put policy API.
```
Comment (jrodewig) [inline docs/reference/ilm/getting-started-ilm.asciidoc]:
```suggestion
and added through the {ilm-init} put policy API. 
```

File: docs/reference/ilm/getting-started-ilm.asciidoc [hunk 17]
```diff
@@ -1,79 +1,93 @@
...
         "actions": {
-          "delete": {}              <6>
+          "delete": {}              <4>
         }
       }
     }
   }
 }
 ------------------------
+<1> The `min_age` defaults to `0ms`, so new indices enter the hot phase immediately.
```
Comment (jrodewig) [inline docs/reference/ilm/getting-started-ilm.asciidoc]: Phases are enclosed in backticks earlier.
```suggestion
<1> The `min_age` defaults to `0ms`, so new indices enter the `hot` phase immediately. 
```

File: docs/reference/ilm/getting-started-ilm.asciidoc [hunk 18]
```diff
@@ -1,79 +1,93 @@
...
+          "delete": {}              <4>
         }
       }
     }
   }
 }
 ------------------------
+<1> The `min_age` defaults to `0ms`, so new indices enter the hot phase immediately. 
+<2> Trigger the `rollover` action when either of the conditions are met.
+<3> Move the index into the delete phase 90 days after rollover.
```
Comment (jrodewig) [inline docs/reference/ilm/getting-started-ilm.asciidoc]:
```suggestion
<3> Move the index into the `delete` phase 90 days after rollover.
```

File: docs/reference/ilm/getting-started-ilm.asciidoc [hunk 19]
```diff
@@ -1,79 +1,93 @@
...
+<4> Trigger the `delete` action when the index enters the delete phase.
```
Comment (jrodewig) [inline docs/reference/ilm/getting-started-ilm.asciidoc]:
```suggestion
<4> Trigger the `delete` action when the index enters the `delete` phase.
```

File: docs/reference/ilm/getting-started-ilm.asciidoc [hunk 20]
```diff
@@ -1,79 +1,93 @@
...
-The index will be deleted 90 days after it is rolled over.
+See <<_actions>> for the complete list of actions available in each phase.
 
 [float]
 [[ilm-gs-apply-policy]]
-=== Applying a policy to our index
+=== Create an index template to apply the lifecycle policy
+
+To automaticaly apply a lifecycle policy to the new write index on rollover, 
+you specify the policy in the index template used to create new indices.
```
Comment (jrodewig) [inline docs/reference/ilm/getting-started-ilm.asciidoc]:
```suggestion
specify the policy in the index template used to create new indices.
```

File: docs/reference/ilm/getting-started-ilm.asciidoc [hunk 21]
```diff
@@ -90,6 +104,11 @@ PUT _template/datastream_template
 -----------------------
 // TEST[continued]
 
+<1> Apply the template to a new index if its name starts with "datastream-".
```
Comment (jrodewig) [inline docs/reference/ilm/getting-started-ilm.asciidoc]:
```suggestion
<1> Apply the template to a new index if its name starts with `datastream-`. 
```

File: docs/reference/ilm/getting-started-ilm.asciidoc [hunk 22]
```diff
@@ -1,79 +1,93 @@
...
-policy to an index. Since we wish specific settings to be applied to
-the new index created from Rollover, we will set the policy via
-index templates.
+For example, the following request creates a `datastream_template` that is applied to new indices
+whose names match the `datastream-*` index pattern. 
+The template configures two {ilm-init} settings:
 
+* `index.lifecycle.name` specifies the name of the lifecycle policy to apply to all new indices that match 
+the index pattern.  
+* `index.lifecycle.rollover_alias` specifies the index alias that will be rolled over
```
Comment (jrodewig) [inline docs/reference/ilm/getting-started-ilm.asciidoc]:
```suggestion
* `index.lifecycle.rollover_alias` specifies the index alias to be rolled over 
```

File: docs/reference/ilm/getting-started-ilm.asciidoc [hunk 23]
```diff
@@ -100,23 +119,17 @@ DELETE /_template/datastream_template
...
-our policy. The other setting used here is `index.lifecycle.rollover_alias`.
-This setting is required when using a policy containing the rollover
-action and specifies which alias to rollover on behalf of this index.
-The intention here is that the rollover alias is also defined on the index.
+[float]
+[[ilm-gs-bootstrap]]
+=== Bootstrap the initial time-series index
 
-To begin, we will want to bootstrap our first index to write to.
+To get things started, you need to bootstrap an initial index and
```
Comment (jrodewig) [inline docs/reference/ilm/getting-started-ilm.asciidoc]:
```suggestion
To get things started, you need to create an initial index and 
```

File: docs/reference/ilm/getting-started-ilm.asciidoc [hunk 24]
```diff
@@ -131,44 +144,40 @@ PUT datastream-000001
...
-we take advantage of <<aliases-write-index,write indices>>. This is a way
-to define an alias to be used for both reading and writing, with only one
-index being the index that is being written to at a time. Rollover swaps
-the write index to be the new index created from rollover, and sets the
-alias to be read-only for the source index.
+When the rollover conditions are met, the rollover action:
+
+* Creates a new index called `datastream-000002`. 
+This matches the `datastream-*` pattern, so the settings from `datastream_template` are applied to the new index.
+* Designates the new index as the write index and makes the bootstrap index read-only.
```
Comment (jrodewig) [inline docs/reference/ilm/getting-started-ilm.asciidoc]:
```suggestion
* Designates the new index as the write index and makes `datastream-000001` read-only.
```

File: docs/reference/ilm/getting-started-ilm.asciidoc [hunk 25]
```diff
@@ -131,44 +144,40 @@ PUT datastream-000001
...
+
+This process will repeat each time the rollover conditions are met.
```
Comment (jrodewig) [inline docs/reference/ilm/getting-started-ilm.asciidoc]:
```suggestion
This process repeats each time rollover conditions are met. 
```

File: docs/reference/ilm/getting-started-ilm.asciidoc [hunk 26]
```diff
@@ -131,44 +144,40 @@ PUT datastream-000001
...
 
 [source,console]
 --------------------------------------------------
 GET datastream-*/_ilm/explain
 --------------------------------------------------
 // TEST[continued]
 
-The above request will retrieve {ilm-init} execution information for all our
-managed indices.
+The response below shows that the bootstrap index is waiting in the `hot` phase's `rollover` action.
```
Comment (jrodewig) [inline docs/reference/ilm/getting-started-ilm.asciidoc]:
```suggestion
The response below shows that the `datastream-000001` index is waiting in the `hot` phase's `rollover` action.
```

File: docs/reference/ilm/getting-started-ilm.asciidoc [hunk 27]
```diff
@@ -1,79 +1,93 @@
...
 
-The goal of this example is to set up a set of indices that will encapsulate
-the data from a time series data source. We can imagine there is a system
-like {filebeat-ref}[Filebeat] that continuously indexes documents into
-our writing index. We wish to roll over the index after it reaches a size
-of 50 gigabytes, or has been created 30 days ago, and then delete the index
-after 90 days.
+. <<ilm-gs-create-policy, Create a lifecycle policy>> with the {ilm-init} Put policy API.  
+. <<ilm-gs-apply-policy, Create an index template>> to apply the policy to each new index.
+. <<ilm-gs-bootstrap, Bootstrap an index>> as the initial write index.
```
Comment (jrodewig) [inline docs/reference/ilm/getting-started-ilm.asciidoc]:
```suggestion
. <<ilm-gs-bootstrap, Create an initial write index>>.
```

File: docs/reference/ilm/getting-started-ilm.asciidoc [hunk 28]
```diff
@@ -100,23 +119,17 @@ DELETE /_template/datastream_template
...
-The first being `index.lifecycle.name`. This setting will configure
-the "datastream_policy" to the index applying this template. This means
-that all newly created indices prefixed "datastream-" will be managed by
-our policy. The other setting used here is `index.lifecycle.rollover_alias`.
-This setting is required when using a policy containing the rollover
-action and specifies which alias to rollover on behalf of this index.
-The intention here is that the rollover alias is also defined on the index.
+[float]
+[[ilm-gs-bootstrap]]
+=== Bootstrap the initial time-series index
```
Comment (jrodewig) [inline docs/reference/ilm/getting-started-ilm.asciidoc]:
```suggestion
=== Create an initial time-series index
```

File: docs/reference/ilm/getting-started-ilm.asciidoc [hunk 29]
```diff
@@ -131,44 +144,40 @@ PUT datastream-000001
...
 [source,console]
 --------------------------------------------------
 GET datastream-*/_ilm/explain
 --------------------------------------------------
 // TEST[continued]
 
-The above request will retrieve {ilm-init} execution information for all our
-managed indices.
+The response below shows that the bootstrap index is waiting in the `hot` phase's `rollover` action.
+It will remain in this state and {ilm-init} will continue to call `attempt-rollover`
```
Comment (jrodewig) [inline docs/reference/ilm/getting-started-ilm.asciidoc]:
```suggestion
It remains in this state and {ilm-init} continues to call `attempt-rollover` 
```

File: docs/reference/ilm/getting-started-ilm.asciidoc [hunk 30]
```diff
@@ -131,44 +144,40 @@ PUT datastream-000001
...
 
 [float]
 [[ilm-gs-check-progress]]
 === Checking progress
 
-Now that we have an index managed by our policy, how do we tell what is going
-on? Which phase are we in? Is something broken? This section will go over a
-few APIs and their responses to help us inspect our indices with respect
-to {ilm-init}.
+To get status information for {ilm-init}-managed indices, you use the Explain API.
```
Comment (jrodewig) [inline docs/reference/ilm/getting-started-ilm.asciidoc]:
```suggestion
To get status information for {ilm-init}-managed indices, you use the explain API. 
```

**Review by jrodewig — APPROVED** [review]
Left some minor editorial questions/suggestions. Otherwise LGTM.

**Comment by debadair** [comment]
Thanks for the review @andreidan & @jrodewig! 

I stuck with the bootstrap terminology, at least for now. (Though since the term now is mostly associated with the framework, it is potentially problematic.)

**Pull request merged**